crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -157,15 +160,34 @@ fn make_executor(kind: ExecutorKind) -> Box<dyn SystemExecutor> {
 }
 
 /// Chain systems into dependencies
-#[derive(PartialEq)]
-pub enum Chain {
-    /// Run nodes in order. If there are deferred parameters in preceeding systems a
-    /// [`apply_deferred`] will be added on the edge.
-    Yes,
-    /// Run nodes in order. This will not add [`apply_deferred`] between nodes.
-    YesIgnoreDeferred,
-    /// Nodes are allowed to run in any order.
-    No,
+#[derive(Default)]
+pub struct Chain(
+    /// Stores a map from the [`TypeId`] of the [`ScheduleBuildPass`] to the
+    /// corresponding boxed [`ScheduleBuildPass::EdgeOptions`], or None if
+    /// the systems are not chained.
+    Option<BTreeMap<TypeId, Box<dyn Any>>>,
+);
+impl Chain {
+    /// Returns if these systems are chained or not.
+    pub fn is_chained(&self) -> bool {
+        self.0.is_some()
+    }
+    /// Set whether these systems are chained or not.
+    pub fn set_chained(&mut self, chained: bool) {
+        if chained && self.0.is_none() {
+            self.0 = Some(BTreeMap::new());
+        } else {
+            self.0 = None;
+        }
+    }
+    /// Add a dependency option to all dependencies established by this chain.
+    pub fn add_option<T: ScheduleBuildPass>(&mut self, option: T::EdgeOptions) {
```

Comment:
I think this should probably be called `add_config` or something similar to avoid confusion with the `Option` type.